core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -472,111 +362,77 @@ public BasicInvocationStage visitGetAllCommand(InvocationContext ctx, GetAllComm
          return invokeNext(ctx, command);
       }
 
-      int commandTopologyId = command.getTopologyId();
       if (ctx.isOriginLocal()) {
-         int currentTopologyId = stateTransferManager.getCacheTopology().getTopologyId();
-         boolean topologyChanged = currentTopologyId != commandTopologyId && commandTopologyId != -1;
-         if (trace) {
-            log.tracef("Command topology id is %d, current topology id is %d", commandTopologyId, currentTopologyId);
-         }
-         if (topologyChanged) {
-            throw new OutdatedTopologyException("Cache topology changed while the command was executing: expected " +
-                        commandTopologyId + ", got " + currentTopologyId);
-         }
+         CacheTopology cacheTopology = checkTopologyId(command);
+         ConsistentHash ch = cacheTopology.getReadConsistentHash();
 
          // At this point, we know that an entry located on this node that exists in the data container/store
          // must also exist in the context.
```

Comment:
I thought at this point the key should be in the context even if it's not in DC, as long as it's owned by the local node in the read CH?